app/src/main/java/org/simple/clinic/onboarding/OnboardingScreen.kt

```diff
@@ -4,26 +4,57 @@ import android.content.Context
 import android.util.AttributeSet
 import android.widget.RelativeLayout
 import androidx.core.widget.NestedScrollView
+import com.f2prateek.rx.preferences2.Preference
 import com.jakewharton.rxbinding2.view.RxView
+import com.spotify.mobius.First
 import io.reactivex.Observable
+import io.reactivex.disposables.Disposable
+import io.reactivex.rxkotlin.cast
 import kotlinx.android.synthetic.main.screen_onboarding.view.*
+import org.simple.clinic.ReportAnalyticsEvents
 import org.simple.clinic.activity.TheActivity
-import org.simple.clinic.bindUiToController
+import org.simple.clinic.crash.CrashReporter
+import org.simple.clinic.mobius.MobiusDelegate
 import org.simple.clinic.registration.phone.RegistrationPhoneScreenKey
 import org.simple.clinic.router.screen.RouterDirection
 import org.simple.clinic.router.screen.ScreenRouter
 import org.simple.clinic.util.clamp
-import org.simple.clinic.widgets.ScreenDestroyed
-import org.simple.clinic.widgets.UiEvent
+import org.simple.clinic.util.scheduler.SchedulersProvider
+import org.simple.clinic.util.unsafeLazy
 import javax.inject.Inject
+import javax.inject.Named
 
 class OnboardingScreen(context: Context, attributeSet: AttributeSet) : RelativeLayout(context, attributeSet), OnboardingUi {
 
   @Inject
   lateinit var router: ScreenRouter
 
+  @field:[Inject Named("onboarding_complete")]
+  lateinit var hasUserCompletedOnboarding: Preference<Boolean>
+
+  @Inject
+  lateinit var schedulersProvider: SchedulersProvider
+
   @Inject
-  lateinit var controller: OnboardingScreenController
+  lateinit var crashReporter: CrashReporter
+
+  private val delegate by unsafeLazy {
```

Comment:
We really need to look at injecting delegates for screens instead of manually constructing them here. It's a lot of unnecessary boilerplate.

Reply:
As discussed, these will be submodules for every feature. We can defer this debt for later.